tatanka/tatanka.go

```diff
@@ -681,3 +723,53 @@ func (t *Tatanka) clientDisconnected(peerID tanka.PeerID) {
 		tt.Send(note)
 	}
 }
+
+// broadcastRates sends market rates to all fiat rate subscribers every
+// t.fiatRateBroadcastInterval. This method blocks and must be called from a
+// goroutine.
+func (t *Tatanka) broadcastRates() {
+	if !t.fiatOracleEnabled() {
+		return
+	}
+
+	broadcastInterval := time.Duration(t.fiatRateBroadcastInterval) * time.Minute
+	ticker := time.NewTicker(broadcastInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-t.ctx.Done():
+			return
+		case <-ticker.C:
+			t.broadcastFiatRates()
+		}
+	}
+}
+
+// broadcastFiatRates sends fiat rates update to all current subscribers.
+func (t *Tatanka) broadcastFiatRates() {
+	if !t.fiatOracleEnabled() {
+		return // nothing to do
+	}
+
+	t.clientMtx.RLock()
+	defer t.clientMtx.RUnlock()
+	topic := t.topics[mj.TopicFiatRate]
+	if topic == nil {
+		return // no subscribers yet
+	}
+
+	for subject, subscribers := range topic.subjects {
+		rate := t.fiatRateOracle.Rate(t.ctx, string(subject))
+		if rate == 0 {
+			continue
+		}
```

Comment:
I'm wondering how important it is to have per-asset subscriptions. Why not just a topic and everybody gets everything? The updates should be relatively compact.

Reply:
Does it mean the admin is tasked with providing rates for certain assets and everyone gets that?

Or users can request a rate that the admin might not have configured?

Reply:
Everything is still mushy here, but I was thinking that the operator would only provide fiat rates for it's configured assets, i.e. `ConfigFile.Chains`. See e.g. `mj.TatankaConfig` which is populated via `(*Tatanka).assets`.

Reply:
Server specifies what the oracle should fetch data for, users receive everything when there's a new rate: https://github.com/decred/dcrdex/pull/2736/commits/8f9328ff68c76e3f5de337db2d8218be285078eb

Reply:
> Everything is still mushy here, but I was thinking that the operator would only provide fiat rates for it's configured assets, i.e. `ConfigFile.Chains`. See e.g. `mj.TatankaConfig` which is populated via `(*Tatanka).assets`.

Didn't get this comment before my latest push. I could update he config to use values from `ConfigFile.Chains`.

Reply:
Updated in https://github.com/decred/dcrdex/pull/2736/commits/dbe0d1bf43a49d89df60fa84af09b668f505355f